triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala

```diff
@@ -260,16 +260,19 @@ object Server {
                     val inputStream = new ByteArrayInputStream(byteString.toArray)
                     DarReader()
                       .readArchive("package-upload", new ZipInputStream(inputStream)) match {
-                      case Failure(err) => complete((StatusCodes.UnprocessableEntity, err))
+                      case Failure(err) =>
+                        complete(errorResponse(StatusCodes.UnprocessableEntity, err.toString))
                       case Success(encodedDar) =>
                         try {
                           val dar = encodedDar.map {
                             case (pkgId, payload) => Decode.readArchivePayload(pkgId, payload)
                           }
                           addDar(compiledPackages, dar)
-                          complete(s"DAR uploaded, main package id: ${dar.main._1}")
+                          complete(
+                            successResponse(s"DAR uploaded, main package id: ${dar.main._1}"))
```

Comment:
This should probably be changed at some point to a more structured response instead of a simple string but let’s leave it for this PR.

Reply:
Maybe a simple object `{"mainPackageId": ...}` within the `results` field would suffice?